### DISCUSSION TITLE
更新日志 - 20230526

### DISCUSSION THREAD

**Discussion by jjm2473** [istoreos/istoreos#802] [Announcements]
# 系统：
1. Luci界面增加“杂项设置”，可配置CPU调度策略，延迟启动（等待硬盘）等
2. Luci界面增加“风扇控制”，仅部分有风扇接口的设备有效（R2S/R4S/R5S等），且只能配置最低档位的温度。x86平台通常在bios中控制风扇，无法使用此插件

# 应用：
* QuickStart：
  1. 系统信息显示的设备型号，对于x86平台，如果bios没有机型数据，则显示成“x86 Generic”

# 设备：
* rk3568: 
  1. 修复上一版本全新刷机或恢复出厂设置以后LED失效的问题
  2. 增加友善电子的R5C，与R5S共用固件。由于涉及到Bootloader的更新，如果R5C刷过老版本R5S固件，需要全新刷机（例如USB线刷或重做TF卡，恢复出厂设置无效），否则依然识别成R5S
  3. 增加莱茵特的T68M

**Comment by feishi312** [comment]
强！！！

**Reply by whois233** [reply]
这个版本更新之后 zerotier 挂了

**Comment by Famiglistimo-X** [comment]
R5S接风扇了，接的是里面ZH1.5mm的风扇接口，风扇启动温度设成35也还是不转，建议加个开机风扇自检转一会

![IMG_4697](https://github.com/istoreos/istoreos/assets/76557864/5a06045a-afb8-4642-bc27-0e2b2a31193f)
![截屏2023-05-27 12 09 59](https://github.com/istoreos/istoreos/assets/76557864/2aab38a3-9e1f-417c-b328-68bb6ccb89ba)

**Reply by jjm2473** [reply]
友善的都是pwm风扇，如果你的风扇不是pwm，那可能完全没用

**Comment by jjm2473** [comment]
R5S的固件上传错误，启动R5C有问题，等重新上传。新固件的文件名末尾应该是 -squashfs-**combined**.img.gz

**Reply by jjm2473** [reply]
已经重新上传

**Comment by yang4221931** [comment]
r5c 刷了 led 还是失效

**Reply by jjm2473** [reply]
之前的固件发错了文件，下载 -squashfs-combined.img.gz 结尾的，重新刷机

**Reply by yang4221931** [reply]
好的👌谢谢

**Comment by wentianci** [comment]
更新了版本后出现使用opkg install 命令时提示错误 Segmentation fault
之前的版本使用opkg install 命令时就没有问题。
![微信截图_20230527220557](https://github.com/istoreos/istoreos/assets/59164321/c3d59bb5-9e84-430d-a22c-f9307623f75e)

**Reply by jjm2473** [reply]
你这是什么设备，用mips架构的软件包？？

**Reply by jjm2473** [reply]
iStoreOS官方可没发布过mips架构的固件

**Reply by wentianci** [reply]
https://github.com/xiaorouji/openwrt-passwall/releases
https://github.com/xiaorouji/openwrt-passwall/releases/download/4.66-5/passwall_packages_ipk_x86_64.zip
https://github.com/xiaorouji/openwrt-passwall/releases/download/4.66-5/luci-app-passwall_4.66-5_all.ipk
https://github.com/xiaorouji/openwrt-passwall/releases/download/4.66-5/luci-i18n-passwall-zh-cn_4.66-5_all.ipk
X86工控机

**Reply by wentianci** [reply]
EXSI+istoreOS

**Reply by wentianci** [reply]
VirtualBox + istoreOS, 安装上面连接上下载的文件没有问题，
当时也没升级最新版本的istoreOS

**Reply by jjm2473** [reply]
既然你的设备是x86，那怎么会安装mips的包，要么就是你下载错了

**Comment by ssssss87815** [comment]
cpu调度在哪呢?怎么没看到- -

**Reply by jjm2473** [reply]
杂项设置

**Reply by ssssss87815** [reply]
杂项里面没有啊

**Comment by foolish-gd** [comment]
R5C刷上一版固件,直接在线更新最新的-squashfs-combined,基本功能全正常,除了多了一个无用的eth0.wan/lan口错误以外.无其他问题.
更新以后能看到风扇管理,能显示CPU温度,但是型号显示错误.
或者dd if=istoreos**.img of=/dev/block/mmcblk0 完成以后,配置保留,是系统里面是可以看到风扇管理.能看到CPU温度的.这时设备被正常识别为R5C,最完美的状态.
这时如果,恢复出厂设置,风扇管理就被屏蔽了
![pic_20230529070427](https://github.com/istoreos/istoreos/assets/64602758/4023058c-3d84-4eec-843e-46f8a192c6dd)

**Reply by jjm2473** [reply]
r5c本来就没风扇，不显示风扇控制才是正常的

**Reply by jjm2473** [reply]
使用dd命令就是全新刷机，因为你都没限制写的范围，不可能只更新bootloader，肯定是整个固件都更新了。dd命令以后应该马上关机，因为存储器已经被刷写过，当前系统的状态是不正常的，不可以再进行其他操作，马上断电才是正确的做法。

**Reply by foolish-gd** [reply]
> r5c本来就没风扇，不显示风扇控制才是正常的
可是R5C有风扇接口啊.

**Reply by jjm2473** [reply]
哪有风扇接口，官网的原理图以及wiki都没看到风扇接口 https://wiki.friendlyelec.com/wiki/index.php/NanoPi_R5C 。如果你认为有，麻烦给出文档。

**Reply by foolish-gd** [reply]
搞错了.我以为那个2针的接口是风扇用的.实际那是RTC接针

**Comment by yangkaiyv** [comment]
今天也是盼着添加定时监测wan口ip，监测到非公ip后自动重播宽带功能到一天 X(

**Reply by jjm2473** [reply]
自己动手，丰衣足食。没有其他人有非公网ip自动重拨的需求，最多就是断网重拨。

**Reply by yangkaiyv** [reply]
；）好吧已经解决了，就在昨晚。抱歉我并不是专业的，但通过gpt学会了写点简单点sh脚本。包括这次的小功能。不得不说istore是很易用的系统。

**Comment by rayangl** [comment]
请问是否可以调整配置文件，修改风扇的转速，cpu有时候瞬时温度达到设定值后突然回落，造成风扇频繁启停，买的4000转的，隔一会儿就“呜”一下，我按照以下教程尝试修改，没有改成功。
https://www.bilibili.com/read/cv18823116
https://github.com/coolsnowwolf/lede/issues/5681
https://github.com/friendlyarm/friendlywrt/blob/master-v19.07.1/target/linux/rockchip-rk3328/base-files/usr/bin/start-rk3328-pwm-fan.sh

**Reply by jjm2473** [reply]
把风扇启动的温度调高，风扇关停的温度调低啊，后台不是有配置页面了吗？pwm会自动调整转速

**Reply by rayangl** [reply]
配置页面只有启动和停止，可不可以自定义不同的温度，配置不同的转速呢

**Reply by jjm2473** [reply]
pwm本身就会根据温度调整转速，而且对应关系只能在dts里配置，应该没法通过软件配置了。
配置页面不是启动和停止，只是是否应用配置而已。不管你用不用配置页面，风扇都有默认的启动温度

**Comment by WJ-JAY** [comment]
树莓派4，这个功能是不生效的，请问下风扇控制原理是什么，控制的是哪个IO口？

**Reply by jjm2473** [reply]
不是控制io口，是修改温度检测的触发值。io相关的是pwm或者gpio，要在dts里面配置好风扇，并且跟温控关联，跟这个ui无关，ui只修改温控值。

**Comment by libli** [comment]
风扇控制的ui有独立ipk吗？我是自己编译的lede，想加入这个风扇控制的ui。

**Reply by jjm2473** [reply]
ipk没有，ui代码在这里 https://github.com/jjm2473/openwrt-apps/tree/main/luci-app-fan

注意不是有ui就能用的，需要设备dts文件配置好风扇和温控区域的关联，具体参考RK机型的dts，包括rk35xx系列，很多都配置了。

**Comment by ghost** [comment]
大佬请教下，我的GL-iNET MT3000使用了kiddin9的kwrt固件，发现自带了一个Luci-app-fan插件，但是启动温度最高只能设置44度，我想修改更高一点无法修改，我查看/rom/usr/libexec/fan-control发现作者是你，所以想请教一下如何修改启动温度，感谢！

**Reply by jjm2473** [reply]
因为这个插件只会控制第一档风扇，所以这个启动温度的上限其实就是第二档风扇的启动温度。所以你说的要调高第二档风扇的温度才行。第二档风扇温度是dts里面定义的，可以找固件作者确认。
